beacon-chain/sync/subscriber_data_column_sidecar_test.go

```diff
@@ -0,0 +1,84 @@
+package sync
+
+import (
+	"testing"
+
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/cache"
+	dbtest "github.com/OffchainLabs/prysm/v6/beacon-chain/db/testing"
+	doublylinkedtree "github.com/OffchainLabs/prysm/v6/beacon-chain/forkchoice/doubly-linked-tree"
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/state/stategen"
+	"github.com/OffchainLabs/prysm/v6/config/params"
+	"github.com/OffchainLabs/prysm/v6/consensus-types/primitives"
+	"github.com/OffchainLabs/prysm/v6/testing/assert"
+	"github.com/OffchainLabs/prysm/v6/testing/require"
+	"github.com/OffchainLabs/prysm/v6/testing/util"
+)
+
+func TestDataColumnSubnets(t *testing.T) {
```

Comment:
Question: Why in this case could not we use sub tests
```go
t.Run("allDataColumnSubnets returns nil when no validators tracked", func(t *testing.T) {
...
})

t.Run("allDataColumnSubnets returns all subnets logic test", func(t *testing.T) {
...
})
```

instead of a list of test cases?

I think it would be more readable here, since for these 2 tests cases the setup and the test almost totally differ.